app/containers/LoginOrRegister.jsx

```diff
@@ -17,28 +17,27 @@ class LoginOrRegister extends Component {
   constructor(props) {
     super(props);
     this.toggleMode = this.toggleMode.bind(this);
-    this.onLoginSubmit = this.onLoginSubmit.bind(this);
-    this.onRegisterSubmit = this.onRegisterSubmit.bind(this);
+    this.handleOnSubmit = this.handleOnSubmit.bind(this);
   }
 
-  onLoginSubmit() {
-    const { dispatch } = this.props;
-    const email = ReactDOM.findDOMNode(this.refs.email).value;
-    const password = ReactDOM.findDOMNode(this.refs.password).value;
-    dispatch(manualLogin({
-      email,
-      password
-    }));
-  }
+  handleOnSubmit(event) {
+    event.preventDefault();
 
-  onRegisterSubmit() {
-    const { dispatch } = this.props;
+    const { dispatch, user: { isLogin } } = this.props;
     const email = ReactDOM.findDOMNode(this.refs.email).value;
     const password = ReactDOM.findDOMNode(this.refs.password).value;
-    dispatch(signUp({
-      email,
-      password
-    }));
+
+    if (isLogin) {
+      dispatch(manualLogin({
+        email,
+        password
+      }));
+    } else {
+      dispatch(signUp({
```

Comment:
Not every login request is a register request, so maybe you should do more logic check before dispatching it.